src/Validation/Validator/UnionValidator.php

```diff
@@ -38,19 +39,23 @@ protected function doValidation(
             return;
         }
 
-        if ($discriminatorName = $this->schema->getDiscriminatorPropertyName()) {
+        $discriminatorName = $this->schema->getDiscriminatorPropertyName();
+        if ($discriminatorName) {
```

Comment:
This was reported by updated coding standard set from PHP_ _CodeSniffer. It looks for assigns in if(...). There we loose type check posibility and its also hard to read.

That's why these assign are decoupled to line above. The condition is now simple and short.